qa/suites/orch/cephadm/workunits/task/test_cephadm_timeout.yaml

```diff
@@ -0,0 +1,12 @@
+roles:
+- - host.a
+  - mon.a
+  - mgr.a
+  - client.0
```

Comment:
not having any OSDs seems to be causing the test to timeout waiting for the cluster to be healthy
https://pulpito.ceph.com/adking-2023-11-05_03:17:01-orch:cephadm-wip-adk-testing-2023-11-04-1920-distro-default-smithi/7448079/

Reply:
or well, there's something with the OSDs. It might not actually be based on the health, but the test seems to have expected 4 of them to have
```
2023-11-05T06:28:33.187 INFO:teuthology.orchestra.run.smithi087.stdout:Scheduled osd.all-available-devices update...
2023-11-05T06:28:33.567 INFO:tasks.cephadm:Waiting for 4 OSDs to come up...
```

Reply:
There's special behavior that use the `--all-available-devices` flag and then expects us to get the number of osds equal to the number of scratch devices it found
```
if cur == 0:
            _shell(ctx, cluster_name, remote, [
                'ceph', 'orch', 'apply', 'osd', '--all-available-devices',
            ])
            # expect the number of scratch devs
            num_osds = sum(map(len, devs_by_remote.values()))
            assert num_osds
        else:
            # expect the number of OSDs we created
            num_osds = cur
```
where the number of devices comes from a call to the teuthology library
```
devs_by_remote[remote] = teuthology.get_scratch_devices(remote)
```

Reply:
for now, added an osd to the roles to see if that fixes things. Can maybe look later into how the number of devices is calculated and why no OSDs are actually getting placed anyway.